### PR TITLE
KAFKA-17624: Remove the E2E uses of accessing ACLs from zk

### PR DESCRIPTION
**Pull request by frankvicky**
JIRA: [KAFKA-17624](https://issues.apache.org/jira/browse/KAFKA-17624)

Delete the usage of `list_acls` and itself.
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Review by chia7712 — COMMENTED** [review]
@frankvicky thanks for this patch

Could you please cleanup the usage of `kafka_acls_cmd_with_optional_security_settings`?

File: tests/kafkatest/tests/core/zookeeper_security_upgrade_test.py [hunk 1]
```diff
@@ -85,28 +85,3 @@ def run_zk_migration(self):
         # restart broker with zookeeper.set.acl=true and acls
         self.kafka.zk_set_acl = True
         self.kafka.restart_cluster()
-
-    @cluster(num_nodes=9)
-    @matrix(security_protocol=["PLAINTEXT", "SSL", "SASL_SSL", "SASL_PLAINTEXT"])
-    def test_zk_security_upgrade(self, security_protocol):
```
Comment (chia7712) [inline tests/kafkatest/tests/core/zookeeper_security_upgrade_test.py:91]: this is the only test case in this file, so please remove this whole file directly

**Review by chia7712 — COMMENTED** [review]
@frankvicky thanks for this patch.

**Comment by chia7712** [comment]
@frankvicky Could you please merge trunk to verify the fix of #17384? The `small` label should be removed.

**Comment by frankvicky** [comment]
It works 😄

**Comment by TaiJuWu** [comment]
@frankvicky Thanks for verification.

**Review by chia7712 — APPROVED** [review]

**Pull request merged**